lib/grape/dsl/request_response.rb

```diff
@@ -157,9 +157,9 @@ def represent(model_class, options)
 
         def extract_with(options)
           return unless options.key?(:with)
-          with_option = options[:with]
+          with_option = options.delete(:with)
```

Comment:
You probably want to `dup` options first, since you're modifying the original hash.